src/OrchardCore.Modules/OrchardCore.ContentLocalization/Sitemaps/LocalizedContentItemsQueryProvider.cs

```diff
@@ -61,41 +62,31 @@ public async Task GetContentItemsAsync(ContentTypesSitemapSource source, Content
 
             if (contentType.Parts.Any(ctd => string.Equals(ctd.Name, nameof(LocalizationPart), StringComparison.Ordinal)))
             {
-                // Get all content items here for reference. Then reduce by default culture.
-                // We know that the content item should be localized.
-                // If it doesn't have a localization part, the content item should have been saved.
-                var queryResults = await _session.Query<ContentItem>()
-                     .With<ContentItemIndex>(ci => ci.ContentType == source.LimitedContentType.ContentTypeName && ci.Published)
-                     .OrderBy(ci => ci.CreatedUtc)
-                     .With<LocalizedContentItemIndex>()
-                     .ListAsync();
-
                 // When limiting items Content item is valid if it is for the default culture.
                 var defaultCulture = await _localizationService.GetDefaultCultureAsync();
 
-                // Reduce by default culture.
-                var items = queryResults
-                    .Where(ci => string.Equals(ci.As<LocalizationPart>().Culture, defaultCulture, StringComparison.Ordinal))
-                    .Skip(source.LimitedContentType.Skip)
-                    .Take(source.LimitedContentType.Take);
-
-                queryContext.ContentItems = items;
-
-                // Provide all content items with localization as reference content items.
-                queryContext.ReferenceContentItems = queryResults
-                    .Where(ci => ci.Has<LocalizationPart>());
+                // Get all content items here for reference. Then reduce by default culture.
+                // We know that the content item should be localized.
+                // If it doesn't have a localization part, the content item should have been saved.
+                contentItems = await session.Query<ContentItem>()
+                    .With<ContentItemIndex>(ci => ci.ContentType == source.LimitedContentType.ContentTypeName && ci.Published)
+                    .OrderBy(ci => ci.CreatedUtc)
+                    .ThenBy(ci => ci.Id)
+                    .With<LocalizedContentItemIndex>(x => x.Culture == defaultCulture)
+                    .Take(take ?? 0)
+                    .Skip(skip ?? 0)
+                    .ListAsync();
             }
             else
             {
                 // Content type is not localized. Produce standard results.
-                var queryResults = await _session.Query<ContentItem>()
+                contentItems = await session.Query<ContentItem>()
                     .With<ContentItemIndex>(x => x.ContentType == source.LimitedContentType.ContentTypeName && x.Published)
                     .OrderBy(x => x.CreatedUtc)
-                    .Skip(source.LimitedContentType.Skip)
```

Comment:
Is that a different configurable limit than the local skip/take?

Reply:
Got it, I see it's passed in skip/take now.

Reply:
Yes. As far as I understand it, it is a limit that you can set on how many items can be in one sitemap along with a starting point.